data/lib/index.js

```diff
@@ -0,0 +1,139 @@
+import { pipe } from 'it-pipe'
+import { batch } from 'streaming-iterables'
+import { CID } from 'multiformats/cid'
+
+import { MAX_BATCH_GET_ITEMS } from '../tables/constants.js'
+import { createCarTable } from '../tables/car.js'
+import { createFerryTable } from '../tables/ferry.js'
+import { createAggregateService } from './aggregate-service.js'
+
+/**
+ * @typedef {import('../types.js').FerryOpts} FerryOpts
+ * @typedef {import('../types.js').CarItem} CarItem
+ * @typedef {import('../types.js').CarItemFerry} CarItemFerry
+ *
+ * @typedef {object} FerryCtx
+ * @property {string} region
+ * @property {string} tableName
+ * @property {FerryOpts} [options]
+ * 
+ * @typedef {object} CarTableCtx
+ * @property {string} region
+ * @property {string} tableName
+ * @property {import('../types.js').CarOpts} [options]
+ */
+
+/**
+ * Add cars to a loading ferry.
+ *
+ * @param {CarItemFerry[]} cars
+ * @param {FerryCtx} ferryCtx
+ */
+export async function addCarsToFerry (cars, ferryCtx) {
+  const ferryTable = createFerryTable(ferryCtx.region, ferryCtx.tableName, ferryCtx.options)
+
+  const ferryId = await ferryTable.getFerryLoading()
+  await ferryTable.addCargo(ferryId, cars)
+
+  return {
+    id: ferryId
+  }
+}
+
+/**
+ * Sets current Ferry as ready if not previously done
+ *
+ * @param {string} ferryId
+ * @param {FerryCtx} ferryCtx
+ */
+export async function setFerryAsReady (ferryId, ferryCtx) {
+  const ferryTable = createFerryTable(ferryCtx.region, ferryCtx.tableName, ferryCtx.options)
+
+  // Update state of ferry to ready
+  try {
+    await ferryTable.setAsReady(ferryId)
+  } catch (/** @type {any} */ error) {
```

Comment:
`setAsReady` IMHO should not be opinated to catch error if it fails to set as ready because its `ConditionExpression` fails... It will only succeed if `- ferry stat is in loading state` and its `size is enough` (this will likely need to change with commD calc).

Therefore, the consumer of this (lambda function calling this lib function `setFerryAsReady`) is the one that should decide if this is a ok error to tolerate. Given consumer of this comes from DynamoDB event streams that can run in parallel, it is possible that multiple calls could happen concurrently. So, first would succeed and second fail because state would not be `loading` anymore. And consumer is ok with this failure and won't need to retry it.